src/main/java/com/greenfoxacademy/goddesstribesbackend/services/BuildingService.java

```diff
@@ -41,4 +53,46 @@ public Mine saveMineAtStart(Kingdom kingdom) {
     return null;
   }
 
+  public ArrayList<Building> findAll(){
+    return buildingRepository.findAll();
+  }
+
+  public ArrayList<Building> findBuildingsByKingdom_Id(Long id){
+    return buildingRepository.findBuildingsByKingdom_Id(id);
+  }
+
+  public BuildingsDTO createBuildingsDTO(String username){
+    List<BuildingDTO> buildingDTOList = new ArrayList<>();
+    if (kingdomRepository.findKingdomByUser_Username(username).isPresent()){
+      Kingdom kingdom = kingdomRepository.findKingdomByUser_Username(username).get();
+      ArrayList<Building> buildingList = findBuildingsByKingdom_Id(kingdom.getId());
+      for (Building building: buildingList){
+        BuildingDTO  buildingDTO = new BuildingDTO();
+        buildingDTO.setId(building.getId());
+
+        if (building instanceof Townhall){
```

Comment:
The `instanceof` usually marks a code smell, and we try to avoid it whenever it is possible. We should think about some other way around, maybe store the type in the object or something.

Reply:
Ok, akkor készítek külön repositoryt minden épülettípusnak, és csak azt az épülettípust veszi be. A JPA ezt lekezeli akkor.

Reply:
Betettem egy buildingType fieldet a Building classba.